### PR TITLE
Add WAN communition over TCP tutorial

### PR DESCRIPTION
**Pull request by rsanchez15**
Generated docs: https://eprosima-dds-router.readthedocs.io/en/docs-tutorial-wan-tcp/

### CODE REVIEW

File: docs/rst/use_cases/wan_tcp.rst [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+1.  **DDS Router**.
+    *eProsima DDS Router* is an end-user software application that enables the connection of distributed DDS networks.
+    That is, DDS entities such as publishers and subscriber deployed in one geographic location and using a dedicated local network will be able to communicate with other DDS entities deployed in different geographic areas on their own dedicated local networks as if they were all on the same network through the use of *DDS Router*.
+
+    This example presents two routers that enable Internet communication:
+
+    * *DDS Router Net A*. This is the DDS Router that is deployed on *LAN A*. This way it is possible for the robot to communicate out-of-the-box with an external DDS Router configured as server.
+    * *DDS Router Net B*. It plays the server role in the communication. It will expose a public network address to which client DDS Routers connect to establish communication.
+
```
Comment (jparisu) [inline docs/rst/use_cases/wan_tcp.rst:42]: Specify that there is no relation between Publisher/Subscriber and client/server. This has been a common mistake of users using TCP example.
You can also comment that communication goes in both directions.

File: docs/rst/use_cases/wan_tcp.rst [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+In this case, we have two participants:
+
+    *   ``local``: this is a simple participant that communicates with all DDS entities it finds in the default DDS Domain, domain 0.
+        For more information about this participant please refer to the :ref:`Simple Participant section <user_manual_participants_simple>` of this documentation.
+
+    *   ``wan``: it is a participant designed for WAN communication or the communication between two *DDS Routers*.
+        It uses the `Fast DDS Initial Peers configuration <https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#initial-peers>`_ to establish a point-to-point communication between two DDS entities, two *DDS Routers* in this case.
+        For now it is not necessary to know more about the details of this configuration as you only need to configure the connection address of the DDS Router.
+
+    For the *DDS Router Net A*, a connection address shall be defined which must be the same as the one exposed by the *DDS Router Net B*.
```
Comment (jparisu) [inline docs/rst/use_cases/wan_tcp.rst:125]: If this is supposed to be under list tag `wan`, it is wrongly formatted

File: docs/rst/use_cases/wan_tcp.rst [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+To finish this step, run the DDS Router with the configuration file created as an argument.
+
+.. code-block::
+
+    docker run -it -v <path/to/file>/ddsrouter_net_A.yaml:/ddsrouter_net_A.yaml ubuntu-fastdds-suite:<FastDDS-Version>
+    ddsrouter -c /ddsrouter_net_A.yaml
+
+The following figure summarizes the deployment on LAN A.
+
+.. figure:: /rst/figures/use_cases/wan_tcp_deployment_net_A.png
```
Comment (jparisu) [inline docs/rst/use_cases/wan_tcp.rst:145]: Change if possible the direction of the arrow so it points Subscriber

File: docs/rst/use_cases/wan_tcp.rst [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+The image below describes the scenario presented in this tutorial.
+
+.. figure:: /rst/figures/use_cases/wan_tcp.png
+   :align: center
+
+Several key elements can be observed in it:
+
+1.  **Fast DDS Subscriber** and **Fast DDS Publisher**.
+    The application used for this tutorial is the *BasicHelloWorldExample*.
+    The *BasicHelloWorldExample* is a Fast DDS application that implements a subscriber and a publisher, to which a configuration of basic DDS and QoS parameters such as DDS domain, transport, communication reliability, among others, can be applied.
```
Comment (jparisu) [inline docs/rst/use_cases/wan_tcp.rst:32]: `QoS` must be added to spelling wordlist

**Review by jparisu — CHANGES_REQUESTED** [review]
Small suggestions

File: docs/rst/use_cases/wan_tcp.rst [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+More specifically, we will run a DDS Subscriber on a machine deployed on a local ``network A`` and a DDS Publisher on a machine deployed on a local ``network B``.
+Both DDS entities will communicate over the Internet by means of the DDS Router.
+
+.. _warning_lan:
+
+.. warning::
+
+    This tutorial is intended for WAN communication.
+    However, if there is only access to a LAN communication, it is possible to follow the tutorial by changing the DDS Domain Id so that the Subscriber uses the default Domain (``0``) and the Publisher uses DDS Domain ``1``.
+    This way the DDS entities are logically isolated and will not discovery other entities out of their DDS Domain.
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/rst/use_cases/wan_tcp.rst:21]: I used the word entities instead of nodes

File: docs/rst/use_cases/wan_tcp.rst (hunk 2 above)
Comment (rsanchez15) [reply docs/rst/use_cases/wan_tcp.rst:125]: Wrongly formated.

**Review by jparisu — APPROVED** [review]
LGTM with green CI

**Pull request merged**